threadbare/operations.py

```diff
@@ -2,17 +2,106 @@
 import contextlib
 import subprocess
 import getpass
-from pssh.clients.native import SSHClient as PSSHClient
 from pssh import exceptions as pssh_exceptions
 import os, sys
 from threadbare import state
-from threadbare.common import merge, subdict, rename, cwd
+from threadbare.common import merge, subdict, rename, cwd, sudo_wrap_command, pwd_wrap_command, shell_wrap_command
+import pssh.clients.native
+
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+# gevent is used by Parallel-SSH which interferes with Python multiprocessing
+# it causes 'things' to block indefinitely
+from gevent import monkey
+monkey.patch_all(thread=False)
```

Comment:
**this** is what has alluded me all day and is the cause of the mysterious hanging behaviour.

parallel-ssh, even when using the non-parallel client is still initialising gevent behind the scenes and apparently there is [a conflict](https://stackoverflow.com/questions/8678307/gevent-monkeypatching-breaking-multiprocessing) between gevent and multiprocessing

Reply:
there was an issue with this preventing `boto` from running (which uses `futures`). removing `thread=False` fixed it